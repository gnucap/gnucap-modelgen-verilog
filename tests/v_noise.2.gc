load ../vams/resistor.so
verilog

module main(1,2,3,4)
resistor #(.r(1.), .$mfactor(1)) r0(1, 0);
resistor #(.r(1.), .noisy(1.)) r1(2, 0);

resistor #(.r(1.), .noisy(1.)) r2(3, 0);
resistor #(.r(1.), .noisy(1.)) r3(3, 0);

resistor #(.$mfactor(2), .r(1.), .noisy(1.)) r4(4, 0);
endmodule
main m(1,2,3,4)

noise_probe #(.name("thermal")) noise1(1);
noise_probe #(.name("thermal")) noise2(2);
noise_probe #(.name("thermal")) noise3(3);
noise_probe #(.name("thermal")) noise4(4);

probe ac npwr(noise*) x(m.r1.*) x(m.r0.*)

list

// BUG: need op
op

ac 1 8 * 2

options temp=300
// BUG: need op
op
ac 1 8 * 2

status notime
end
