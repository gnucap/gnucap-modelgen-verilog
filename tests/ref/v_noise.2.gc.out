module main (1,2,3,4);
  resistor #(.$mfactor(1),.r(1.)) r0 (1,0);
  resistor #(.noisy(1.),.r(1.)) r1 (2,0);
  resistor #(.noisy(1.),.r(1.)) r2 (3,0);
  resistor #(.noisy(1.),.r(1.)) r3 (3,0);
  resistor #(.noisy(1.),.r(1.),.$mfactor(2)) r4 (4,0);
endmodule // main

main #() m (.1(1),.2(2),.3(3),.4(4));
meas_noise #(.name("thermal")) noise1 (.p(1));
meas_noise #(.name("thermal")) noise2 (.p(2));
meas_noise #(.name("thermal")) noise3 (.p(3));
meas_noise #(.name("thermal")) noise4 (.p(4));
#          
 27.       
#Freq       npwr(noise1) npwr(noise2) npwr(noise3) npwr(noise4) x(m.r1._b_p_n) x(m.r1._b_white_noise_0_p_white_noise_0_n) x(m.r0._b_p_n) x(m.r0._b_white_noise_0_p_white_noise_0_n)
 1.         0.         17.E-21    8.E-21     8.E-21     ??         ??         ??         ??        
 2.         0.         17.E-21    8.E-21     8.E-21     ??         ??         ??         ??        
 4.         0.         17.E-21    8.E-21     8.E-21     ??         ??         ??         ??        
 8.         0.         17.E-21    8.E-21     8.E-21     ??         ??         ??         ??        
#          
 300.      
#Freq       npwr(noise1) npwr(noise2) npwr(noise3) npwr(noise4) x(m.r1._b_p_n) x(m.r1._b_white_noise_0_p_white_noise_0_n) x(m.r0._b_p_n) x(m.r0._b_white_noise_0_p_white_noise_0_n)
 1.         0.         32.E-21    16.E-21    16.E-21    ??         ??         ??         ??        
 2.         0.         32.E-21    16.E-21    16.E-21    ??         ??         ??         ??        
 4.         0.         32.E-21    16.E-21    16.E-21    ??         ??         ??         ??        
 8.         0.         32.E-21    16.E-21    16.E-21    ??         ??         ??         ??        
Gnucap   System status
iterations: op=2, dc=0, tran=0, fourier=0, total=13
transient timesteps: accepted=0, rejected=0, total=0
nodes: user=4, subckt=0, model=0, total=4
dctran density=25.0%, ac density=25.0%
